platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/TaskScheduler.java

```diff
@@ -71,12 +81,20 @@ protected TaskScheduler(
             @NonNull final String name,
             @NonNull final TaskSchedulerType type,
             final boolean flushEnabled,
+            final boolean squelchingEnabled,
             final boolean insertionIsBlocking) {
 
         this.model = Objects.requireNonNull(model);
         this.name = Objects.requireNonNull(name);
         this.type = Objects.requireNonNull(type);
         this.flushEnabled = flushEnabled;
+
+        if (squelchingEnabled) {
+            this.squelcher = new DefaultSquelcher();
+        } else {
+            this.squelcher = new NoopSquelcher();
```

Comment:
Im wondering if maybe we should throw an exception instead of a noop. if you try to squelch something that wont do it, this would mean we silently ignore it

Reply:
i now see that it does throw. maybe we rename it to something like `ThrowingSquelcher`

Reply:
Done